src/Core/MainLoop.js

```diff
@@ -92,8 +92,26 @@ MainLoop.prototype._step = function _step(view, timestamp) {
 
     // update camera
     const dim = this.gfxEngine.getWindowSize();
+    const foo = (dim.width != view.camera.width || dim.height != view.camera.height);
+
+    const animator = {
+        t: 0,
+        update: (dt, ignore) => {
+            this.t = ignore ? 0 : Math.min(1, dt / 1000 + this.t);
+
+            if (this.t < 1) {
+                if (foo) {
+                    view.controls.updateCamera();
```

Comment:
`view.controls` exists only for `GlobeView`.

`view.controls.updateCamera` exists only for GlobeControls. On overall, this is a GlobeControls issue. It should be in GlobeControls. Actually, I think GlobeControls should be a registered FrameRequester and this should happen in its update method.

Reply:
And I think `foo` calculation should be moved inside the frame requester to ease this.